pkg/api/v1/defaults.go

```diff
@@ -160,6 +160,18 @@ func SetDefaults_Pod(obj *Pod) {
 			}
 		}
 	}
+	for i := range obj.Spec.InitContainers {
+		if obj.Spec.InitContainers[i].Resources.Limits != nil {
```

Comment:
Yes, that's current behavior. One case in my mind is that: for the Pod, if request is not set, it is low priority (not defined in QoS??) which maybe killed. So set request for Pod to avoid eviction; it's OK to RC/RS which will start a new Pod.